### ISSUE TITLE
Finish documentation

### ISSUE THREAD

**Issue by shaunco** [adelyte/crescendo#32]
I'm pretty new to SIMPL, but I've created a few touchscreen interfaces in VT and the logic in SIMPL, yet I really can't make heads or tails of Crescendo...  It would be great if the documentation was completed, or at a minimum, fix the link to the "demo program" at https://www.adelyte.com/crestron/crescendo/docs/basics/getting-started along with details on configuring rooms and sources.

**Comment by shaunco** [comment]
Issue #22 pointed me to Crescendo MC3.smw.  I should have realized that it was the only program in the program folder, whereas everything else was a user module.  Hopefully I'll get much further, but it would still be great to have complete documentation.

**Comment by adelyte-chris** [comment]
I've fixed the download link on the _Getting Started_ page. Thank you for the catch.

The framework is provided as a standard program archive in `/program` with the other two directories being self-explanatory. We expected developers to be familiar with that flat file hierarchy, but obviously more explanation is warranted.

We will add a _Finding Your Way Around_ section to the _Getting Started_ page.

**Comment by nmelin** [comment]
Is there a single reference for all crescendo namespaces (all models, collections, interfaces, etc.), including commonly used namespace for custom functions that don't exist in the framework template?

Are there any namespace changes that are currently under consideration?  

We're seeking to synchronize program namespaces with system engineering documentation and a single reference would be very helpful, as would awareness of any upcoming changes.  Scouring the documentation site yielded no answers.

Perhaps this should be a separate issue?

Thanks  @adelyte-austin @adelyte-chris

**Comment by adelyte-chris** [comment]
By “namespace” do you mean “module” or are you thinking more like
subfolders in the SIMPL Windows program? (I’m not trying to be pedantic, I
want to make sure I answer your question.)

Would you provide an example of a custom function?

**Comment by nmelin** [comment]
Hi Chris, I'm thinking in the simpl windows program, signal naming conventions.  

Namespace examples, where  name = 
<namespace identifier> separator <local name>


Sources signal names =    S-{id_number}__
Lights signal names =    L-{id_number}__
Switchers signal names =    X-{id_number}__

Is there better nomenclature to use for this than 'namespace'? 

Sent from my iPhone

On Dec 28, 2016, at 12:01 PM, Chris Massey <notifications@github.com> wrote:

By “namespace” do you mean “module” or are you thinking more like
 in the SIMPL Windows program? (I’m not trying to be pedantic, I
want to make sure I answer your question.)

Would you provide an example of a custom function?

**Comment by adelyte-chris** [comment]
We used “namespace” in our [original documentation](https://code.google.com/archive/p/crescendo-framework/wikis) on Google Code. It seems like a fine term to me.

Would something like our old [SIMPL Style Guide](https://code.google.com/archive/p/crescendo-framework/wikis/SimplStyle.wiki) updated with our best practices suffice?

We typically define global subsystems as `Z` and highly specific custom subsystems as `Y` if they don't fit `D` (Devices), `C` (Climate), `W` (Shades), or `L` (Lighting). For example, a standalone driveway sensor might be `Y-01` because it's not a device—it offers no direct control.

**Comment by nmelin** [comment]
Yes an updated SIMPL style guide would be helpful.  For the purposes on engineering documentation, our team would need to expand the prefixes to cover processors, controllers, gateways, sensors, etc, instead of grouping them into D or Y.

Reference for others, pulled from [SIMPL Style Guide](https://code.google.com/archive/p/crescendo-framework/wikis/SimplStyle.wiki)

| Prefix | Module | 
------------ | -------------
| R | Room | 
| A | Audio Zone | 
| V | Video Zone |
| X | Switcher | 
| S | Source | 
| L | Lights | 
| W | Shades | 
| C | Climate | 
| D | Other | 
| M | Menu | 
| T | Touchpanel | 
| H | Remote | 
| K | Keypad | 
| Z | System |

**Comment by adelyte-chris** [comment]
Note that there is no such thing as `M` (menu) in Crescendo 2.

Is a “processor” a Crestron processor? If so, then use `Z` as follows:
`Z-00` (this processor, contextual to each program), `Z-01` to `Z-99` (a
processor with a given ID). For example, a heartbeat signal from processor
3 would be `Z-03__Heartbeat!`.

For gateways, use `G` (gateway).

For sensors, use `E` (sensor event). For example, occupancy sensor 1 motion
detected might be `E-01__Motion_Detected?`.

Picking letters is almost arbitrary. Propose some and we’ll commit to them.
